### ISSUE TITLE
Can't Get it to Run

### ISSUE THREAD

**Issue by Aezaeo** [Dracovian/Discord-Scraper#4]
Hello, 

First off, I'd like to state that I am no programmer at all just a user. With that being said I followed your instructions to try to get this working but when I run it nothing happens.

I have the latest versions of both python2 and python3 installed. As for my token I don't use Two-factor Authentication for my discord so I found a way to get my Authorization token via Develeoper Panel>Network. I am unsure if this will work or not. This could be my main issue.

I just tested a single server/channel just to see how it worked but as I stated nothing happened. Here's how my config.json looks:
`{
	"token":		 "mfa.<MY-TOKEN>",
	"servers":		 {
		"<103547166454198272>": [
			"<227314046607556609>"
		]
	}
}`

Let me know if you see anything wrong with that.

I'm using the drag and drop method from windows cmd to try to open file, i.e. drag python2_discord.py to cmd window. This could also be my issue? As I said I'm no programmer and am completely new to this.

When dragging "python2_discord.py" I get this error:
`Traceback (most recent call last):
  File "F:\Pictures\Image Downloader\Discord-Scraper\python2_discord.py", line 14, in <module>
    from urllib2 import build_opener, install_opener, urlopen, HTTPError
ModuleNotFoundError: No module named 'urllib2'`

When dragging "python3_discord.py" I just get a quick cmd window flash then nothing. Same thing happens with a double click of the .py file, quick flash and nothing. 

With "python2_discord.py" no Discord Scrapes folder is being created but with "python3_discord.py" it is being created. But nothing is being downloaded and again as I stated the .py file just opens then closes right away.

Thanks.

**Comment by Aezaeo** [comment]
@Dracovian That did it, it's working now, thanks for the help. Another question, after downloading all the images from a channel, if new images were to be posted, if I reran the script, would it only download the new images or would it scan the entire channel again and try to determine duplicates?

**Comment by Dracovian** [comment]
It will scan the entire channel again, I could implement a function that creates a dump of all existing images in the folder and force the scraper to skip those in favor of grabbing new images.

**Comment by Dracovian** [comment]
I've recently updated the scripts to fix that issue.

My first notion with the script was that I noticed how Discord would allow for multiple images to have the same filename (i.e. "image.jpg") and I didn't pay much attention to the subfolder where the image was uploaded. So the way around having images with different content despite having similar filenames was to alter the filenames to something that is unlikely to produce duplicates (but there was always the slight chance of the random generator producing duplicate results).

So I decided to fix the scripts by removing the random generator and opting to prepend the subfolder name onto the filename which fixes the issue of having different images with similar filenames without creating a bunch of duplicates (not as many duplicates).
